providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -172,6 +174,9 @@ private void handleStateTransition(ProviderState oldState, ProviderState newStat
         if (ProviderState.READY.equals(oldState) && ProviderState.READY.equals(newState)) {
             log.debug("Configuration changed");
             ProviderEventDetails details = ProviderEventDetails.builder().message("configuration changed").build();
+            if (!changedFlagKeys.isEmpty()) {
```

Comment:
I'm just curious about the benefit of this. Can we add an empty list of changed flags? In my brain, this should work the same way if I pass changed flags or an empty list of changed flags. I have to admit that I don't have the complete picture, nor am I aware of all the implications  - but ideally, we can remove this condition and reduce code complexity

Reply:
I agree by defining an EMPTY_LIST in the constructor we are eliminating null pointers, and empty list shouldn't cause problems, will implement this

Reply:
Although in GRPC connector, when we were passing values to state consumer, I was passing it as null. Have also made the change to pass an EmptyList instead to avoid null pointer exception and bring null anywhere